Source/OCMock/OCPartialMockObject.m

```diff
@@ -131,6 +153,16 @@ - (void)handleUnRecordedInvocation:(NSInvocation *)anInvocation
 	}
 }
 
+- (void)verifyInvocation:(OCMInvocationMatcher *)matcher withQuantifier:(OCMQuantifier *)quantifier atLocation:(OCMLocation *)location
+{
+    SEL selector = [[matcher recordedInvocation] selector];
+    if (![self hasSetupForwarderForSelector:selector])
+    {
+        [NSException raise:NSInvalidArgumentException
+            format:@"Selector `%@` can not be verified using OCMVerify(). Use OCMExpect() instead.", NSStringFromSelector(selector)];
```

Comment:
I like the idea, but this is not really correct. Some methods that can't be used in `verify` also cannot be used in `expect`. And, additionally, as the documentation points out, sometimes it is possible to stub a method and then verify it.

The reason is simple: at some point it got too complex to determine for which methods on `NSObject` the forwarder could be installed. So all of them are skipped. However, when a stub is added in `-addStub:`, the mock sets up the forwarder, which in almost all cases the developers care about actually works.

Reply:
Ack. That just requires that the developer add the stub which wasn't happening here. They just wanted to verify that the routine got called. I've pushed a new version that I am not 100% happy with (the "addedInstructions" is really hacky) but I think at least gives an indication to developers how to possibly fix the problem they are seeing.